lib/glific/wa_managed_phones.ex

```diff
@@ -183,4 +207,43 @@ defmodule Glific.WAManagedPhones do
   @spec has_any_phones(list()) :: {:ok, list()} | {:error, String.t()}
   defp has_any_phones([]), do: {:error, "No active phones available"}
   defp has_any_phones(wa_managed_phones), do: {:ok, wa_managed_phones}
+
+  @doc """
+  add the phone status
+  """
+  @spec status(String.t(), non_neg_integer()) :: {:ok, WAManagedPhone} | {:error, String.t()}
+  def status(new_status, phone_id) do
+    organization_id = Repo.get_organization_id()
+
+    with %WAManagedPhone{} = phone <- get_wa_managed_phone(phone_id),
+         {:ok, wa_managed_phone} <-
+           phone
+           |> WAManagedPhone.changeset(%{status: new_status})
+           |> Repo.update() do
+      if new_status not in ["active", "loading"] do
+        Notifications.create_notification(%{
+          category: "WhatsApp Groups",
+          message:
+            "Cannot send messages. WhatsApp phone #{wa_managed_phone.phone} is not connected with Maytapi. Current status: #{wa_managed_phone.status}",
+          severity: Notifications.types().critical,
+          organization_id: organization_id,
+          entity: %{
+            phone: wa_managed_phone.phone,
+            status: new_status
+          }
+        })
+      end
+
+      {:ok, wa_managed_phone}
+    else
+      nil ->
+        {:error, "Phone ID not found"}
+
+      {:error, changeset} ->
+        {:error, "Failed to update status: #{inspect(changeset.errors)}"}
+
+      _ ->
+        {:error, "Unexpected error occurred"}
```

Comment:
Is there a case of unexpected error?, like we are handling all possible cases separately

Reply:
no there isn't